File: ЛР_3/LR_3_v31.cpp
#include <iostream>
#include <math.h>
#include <iomanip>

using namespace std;
int main() {
	
	float x, s = 0;
	cin >> x;
	int i = 1;
	float g = 0.000001;
	float a1;
	float a2 = 1;
	
	do{  
		a1 = a2;
		a2 = pow(-1, i) * pow(x, i) / (i + 1);
		s += a1;
		i++;
	}
	while(fabs(a2 - a1) > g);
	cout << fixed << setprecision(6) << s;
}

File: ЛР_3/LR_3_v31.py
from math import pow, fabs

x = float(input())
s = 0
g = 0.000001
i = 1
a1 = 1
a2 = 1
while fabs(pow(-1, i) * pow(x, i) / (i + 1) - pow(-1, i - 1) * pow(x, i - 1) / i) > g:
    s += pow(-1, i - 1) * pow(x, i - 1) / i
    i += 1

print('%.6f' % s)
